services-speech/src/main/java/com/mapbox/api/speech/v1/MapboxSpeech.java

```diff
@@ -0,0 +1,192 @@
+  abstract Cache cache();
+
+  @Nullable
+  abstract String accessToken();
+
+  @Nullable
+  abstract String instruction();
+
+  @Override
+  protected abstract String baseUrl();
+
+  @Override
+  public synchronized OkHttpClient getOkHttpClient() {
+    if (okHttpClient == null) {
+      OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
+      if (isEnableDebug()) {
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(HttpLoggingInterceptor.Level.BASIC);
+        httpClient.addInterceptor(logging);
+      }
+      if (cache() != null) {
+        httpClient.cache(cache());
+      }
+
+      okHttpClient = httpClient.build();
+    }
+    return okHttpClient;
+  }
+
+  /**
+   * Creates a builder for a MapboxSpeech object with a default cache size of 10 MB.
+   *
+   * @return a builder to create a MapboxSpeech object
+   * @since 3.0.0
+   */
+  public static Builder builder() {
+    return new AutoValue_MapboxSpeech.Builder()
+            .baseUrl(Constants.BASE_API_URL);
+  }
+
+  /**
+   * This builder is used to create a MapboxSpeech instance, with details about how the API calls
+   * should be made (input/output format, language, etc.). To use caching, specify a cache
+   * directory. Access token and instruction are required, along with cache directory if you choose
+   * to use caching.
+   *
+   * @since 3.0.0
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /**
+     * Language of which to request the instructions be spoken. Default is "en-us"
+     *
+     * @param language as a string, i.e., "en-us"
+     * @return this builder for chaining options together
+     * @since 3.0.0
+     */
+    public abstract Builder language(String language);
+
+    /**
+     * Format which the input is specified. If not specified, default is text
+     *
+     * @param textType either text or ssml
+     * @return this builder for chaining options together
+     * @since 3.0.0
+     */
+    public abstract Builder textType(String textType);
+
+    /**
+     * Output format for spoken instructions. If not specified, default is mp3
+     *
+     * @param outputType either mp3 or json
+     * @return this builder for chaining options together
+     * @since 3.0.0
+     */
+    public abstract Builder outputType(String outputType);
+
+    /**
+     * Required to call when this is being built. If no access token provided,
+     * {@link ServicesException} will be thrown.
+     *
+     * @param accessToken Mapbox access token, You must have a Mapbox account in order to use
+     *                    the Optimization API
+     * @return this builder for chaining options together
+     * @since 3.0.0
+     */
+    public abstract Builder accessToken(String accessToken);
+
+    /**
+     * @param instruction to dictate
+     * @return this builder for chaining options together
+     * @since 3.0.0
+     */
+    public abstract Builder instruction(String instruction);
```

Comment:
Why not annotate this NonNull and remove the check in `build()`?

Reply:
@cammace that's what I had originally, but looking at the other API's it seems we usually throw our own `ServiceException`, but I can do it either way just let me know

Reply:
Prefer annotations over checks so we reduce executable code where possible.

Reply:
will update that